subsys/net/l2/openthread/openthread.c

```diff
@@ -428,27 +438,27 @@ static int openthread_init(struct net_if *iface)
 		platformShellInit(ot_context->instance);
 	}
 
-	if (IS_ENABLED(CONFIG_OPENTHREAD_NCP)) {
+	if (IS_ENABLED(CONFIG_OPENTHREAD_COPROCESSOR)) {
 		otNcpInit(ot_context->instance);
-	}
-
-	otIp6SetEnabled(ot_context->instance, true);
+	} else {
+		otIp6SetEnabled(ot_context->instance, true);
+
+		if (!IS_ENABLED(CONFIG_OPENTHREAD_COPROCESSOR_NCP)) {
```

Comment:
This check is redundant now, no longer needed.

Reply:
Removed.